packages/api/src/external/carequality/patient.ts

```diff
@@ -25,41 +29,45 @@ import {
   getOutboundPatientDiscoveryRespCount,
   getOutboundPatientDiscoveryResps,
 } from "./command/outbound-patient-discovery-resp/get-outbound-patient-discovery-resp";
-import { createPatientDiscoveryRequest } from "./create-pd-request";
 import { CQLink } from "./cq-patient-data";
-import { OutboundPatientDiscoveryResp } from "./patient-discovery-result";
+import { createPatientDiscoveryRequest } from "./create-pd-request";
 import { cqOrgsToXCPDGateways, generateIdsForGateways } from "./organization-conversion";
-import { MedicalDataSource } from "@metriport/core/external/index";
+import { OutboundPatientDiscoveryResp } from "./patient-discovery-result";
 import { PatientDataCarequality } from "./patient-shared";
-import { getCQGateways } from "./command/cq-directory/cq-gateways";
 
 dayjs.extend(duration);
 
+const context = "cq.patient.discover";
+const iheGateway = makeIheGatewayAPIForPatientDiscovery();
+
+export const PATIENT_DISCOVERY_TIMEOUT = dayjs.duration({ minutes: 0.25 });
+const CHECK_DB_INTERVAL = dayjs.duration({ seconds: 5 });
+
 export function getCQData(
   data: PatientExternalData | undefined
 ): PatientDataCarequality | undefined {
   if (!data) return undefined;
   return data[MedicalDataSource.CAREQUALITY] as PatientDataCarequality; // TODO validate the type
 }
 
-const createContext = "cq.patient.discover";
+export async function discover(patient: Patient, facilityNPI: string): Promise<void> {
+  const baseLogMessage = `CQ PD - M patientId ${patient.id}`;
+  const { log: outerLog } = out(baseLogMessage);
+  const { cxId } = patient;
 
-export const PATIENT_DISCOVERY_TIMEOUT = dayjs.duration({ minutes: 0.25 });
-const CHECK_DB_INTERVAL = dayjs.duration({ seconds: 5 });
+  if (!iheGateway) return outerLog(`IHE GW not available, skipping PD`);
+  if (!(await isCQDirectEnabledForCx(cxId))) {
+    return outerLog(`CQ disabled for cx ${cxId}, skipping PD`);
+  }
```

Comment:
If the IHE GW is not available or CQ not enabled for this cx, just return.

Reply:
why isn't `!iheGateway` an error/programming error case like in DQ?

Reply:
It's expected to DQ for a patient on sandbox where we don't have IHE GW - https://github.com/metriport/metriport/pull/1655#discussion_r1501590918